storage/mysql/mysql.go

```diff
@@ -18,41 +18,54 @@ package mysql
 import (
 	"context"
 	"database/sql"
+	"fmt"
+	"time"
 
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/transparency-dev/merkle/rfc6962"
 	tessera "github.com/transparency-dev/trillian-tessera"
+	"github.com/transparency-dev/trillian-tessera/api"
+	"github.com/transparency-dev/trillian-tessera/storage"
+	"golang.org/x/sync/errgroup"
 	"k8s.io/klog/v2"
 )
 
 const (
 	selectCheckpointByIDSQL         = "SELECT `note` FROM `Checkpoint` WHERE `id` = ?"
 	replaceCheckpointSQL            = "REPLACE INTO `Checkpoint` (`id`, `note`) VALUES (?, ?)"
 	selectSubtreeByLevelAndIndexSQL = "SELECT `nodes` FROM `Subtree` WHERE `level` = ? AND `index` = ?"
+	replaceSubtreeSQL               = "REPLACE INTO `Subtree` (`level`, `index`, `nodes`) VALUES (?, ?, ?)"
 	selectTiledLeavesSQL            = "SELECT `data` FROM `TiledLeaves` WHERE `tile_index` = ?"
+	replaceTiledLeavesSQL           = "REPLACE INTO `TiledLeaves` (`tile_index`, `data`) VALUES (?, ?)"
 
-	checkpointID = 0
+	checkpointID    = 0
+	entryBundleSize = 256
 )
 
 // Storage is a MySQL-based storage implementation for Tessera.
 type Storage struct {
-	db *sql.DB
+	db    *sql.DB
+	queue *storage.Queue
 
-	newCheckpoint tessera.NewCPFunc
+	newCheckpoint   tessera.NewCPFunc
+	parseCheckpoint tessera.ParseCPFunc
 }
 
 // New creates a new instance of the MySQL-based Storage.
 func New(ctx context.Context, db *sql.DB, opts ...func(*tessera.StorageOptions)) (*Storage, error) {
 	opt := tessera.ResolveStorageOptions(nil, opts...)
 	s := &Storage{
-		db:            db,
-		newCheckpoint: opt.NewCP,
+		db:              db,
+		newCheckpoint:   opt.NewCP,
+		parseCheckpoint: opt.ParseCP,
 	}
 	if err := s.db.Ping(); err != nil {
 		klog.Errorf("Failed to ping database: %v", err)
 		return nil, err
 	}
 
+	s.queue = storage.NewQueue(ctx, time.Second, entryBundleSize, s.sequenceBatch)
```

Comment:
Maybe pull out the `time.Second` to a constant with `entryBundleSize` so that these tuning parameters are co-located.

Reply:
There's a storage opt for these BTW - `WithBatching()`
https://github.com/transparency-dev/trillian-tessera/blob/main/log.go#L82

Reply:
Added `defaultEntryBundleSize` and `defaultQueueMaxAge` as the default storage options. `WithBatching()` storage option is also respected.